Python/Product/PythonTools/ptvsd/debugger.py

```diff
@@ -880,8 +861,9 @@ def _stackless_schedule_cb(self, prev, next):
             current.trace_function = current_tf
 
     def trace_func(self, frame, event, arg):
-        # If we're so far into process shutdown that sys is already gone, just stop tracing.
-        if sys is None:
+        # If we're so far into process shutdown that modules are being unloaded, stop tracing
+        # since we can't rely on any modules we've imported to still be working.
+        if sys is None or not sys.modules:
```

Comment:
In 3.6, the shutdown process is subtly different, and `sys is None` doesn't cut it anymore. This fixes issues where you see a bunch of gunk about unhandled exceptions being ignored when deleting objects referenced from module's globals on shutdown (said exceptions coming from our tracefunc).